src/Entity/AbstractValuation.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+namespace Jedkirby\ValPal\Entity;
+
+abstract class AbstractValuation
+{
+    /**
+     * @var string
+     */
+    protected $minValuation;
+
+    /**
+     * @var string
+     */
+    protected $valuation;
+
+    /**
+     * @var string
+     */
+    protected $maxValuation;
+
+    /**
+     * @var string
+     */
+    protected $propertyType;
+
+    /**
+     * @var string
+     */
+    protected $tenure;
+
+    /**
+     * @var string
+     */
+    protected $bedrooms;
+
+    /**
+     * @var string
+     */
+    protected $propertyConstructionYear;
+
+    /**
+     * @var string
+     */
+    protected $minRentalValuation;
+
+    /**
+     * @var string
+     */
+    protected $rentalValuation;
+
+    /**
+     * @var string
+     */
+    protected $maxRentalValuation;
+
+    /**
+     * @param string $value
+     */
+    public function setMinValuation($value)
+    {
+        $this->minValuation = $value;
+    }
+
+    /**
+     * @param string $value
+     */
+    public function setValuation($value)
+    {
+        $this->valuation = $value;
+    }
+
+    /**
+     * @param string $value
+     */
+    public function setMaxValuation($value)
+    {
+        $this->maxValuation = $value;
+    }
+
+    /**
+     * @param string $type
+     */
+    public function setPropertyType($type)
+    {
+        $this->propertyType = $type;
+    }
+
+    /**
+     * @param string $tenure
+     */
+    public function setTenure($tenure)
+    {
+        $this->tenure = $tenure;
+    }
+
+    /**
+     * @param string $bedrooms
+     */
+    public function setBedrooms($bedrooms)
+    {
+        $this->bedrooms = $bedrooms;
+    }
+
+    /**
+     * @param string $year
+     */
+    public function setPropertyConstructionYear($year)
+    {
+        $this->propertyConstructionYear = $year;
```

Comment:
unless  the API handles it, would consider some constraints here, even if it's just that the year "makes sense" (i.e. is 4 digits) - unless again this is a string because it could be a description such as "c.1950" for estimated dates